freqtrade/freqtradebot.py

```diff
@@ -478,9 +478,22 @@ def check_and_call_adjust_trade_position(self, trade: Trade):
 
         if stake_amount is not None and stake_amount < 0.0:
             # We should decrease our position
-            # TODO: Selling part of the trade not implemented yet.
-            logger.error(f"Unable to decrease trade position / sell partially"
-                         f" for pair {trade.pair}, feature not implemented.")
+            # TODO : debug
+            open_sell_order = trade.select_order('sell', True)
+            if open_sell_order:
+                msg = {
+                    'type': RPCMessageType.WARNING,
+                    'status': 'bug open_order_id is None'
+                }
+                self.rpc.send_msg(msg)
+                return
+            # Starategy should return amount in base currency to avoid dust
+            amount = -stake_amount
+            if amount > trade.amount:
+                logger.info("Amount is higher than available.")
+                return
+            self.execute_trade_exit(trade, current_rate, sell_reason=SellCheckTuple(
```

Comment:
Strategy will not return this in base currency - but in stake currency.

Doing it the way you propose would mean that to increase values - the same method (`adjust_trade_position`) would have to return:
* positive values **in stake currency** to increase amount
* negative values **in base currency** to reduce positions

That's a terrible, terrible interface, and very prone to user mistakes - and as such, not an interface i'd accept into freqtrade.

it's freqtrade's (and as PR creator, your) responsibility to convert the stake-amount to the corresponding base amount at the current sell rate.

You can avoid dust by returning slightly more than what you own.

Once the "remaining stake" get's to below minimum tradable amount (usually around 10$ - but it can also be 50$ on some exchanges - use `get_min_pair_stake_amount()` to determine this value)) - you'll have to either reject the partial sell (with "remaining would be too small") - or sell "the remaining bit".

Reply:
Updated